src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -20,17 +21,46 @@ export default class SimidMapCreative extends BaseSimidCreative {
   }
 
   /** @override */
-  onStart(eventData){
-    super.onStart(eventData);
-    //ToDo(juliareichel@): handle invalid JSON and param errors
-    const adParams = JSON.parse(this.creativeData.adParameters);
+  onInit(eventData) {
+    this.handleAdParams_(eventData);
+    super.onInit(eventData);
+  }
+
+  /**
+   * Checks validity of ad parameters and rejects with proper message if invalid.
+   * @param eventData an object that contains information details for a particular event
+   *   such as event type, unique Ids, creativeData and environmentData.
+   * @private 
+  */ 
+  handleAdParams_(eventData) {
+    this.creativeData = eventData.args.creativeData;
+    let adParams = "";
+    if (this.creativeData.adParameters == "") {
+      this.simidProtocol.reject(eventData, {errorCode: CreativeErrorCode.UNSPECIFIED, 
+        message: "Ad parameters not found"});
+    }
+
+    try {
+      adParams = JSON.parse(this.creativeData.adParameters);
+    } catch (exception) {
+      this.simidProtocol.reject(eventData, {errorCode: CreativeErrorCode.CREATIVE_INTERNAL_ERROR, 
+        message: "Invalid JSON input"});
+    }
     const buttonLabel = adParams[AdParamKeys.BUTTON_LABEL]; 
-    //ToDo(juliareichel@): handle case where searchQuery is undefined
     const searchQuery = adParams[AdParamKeys.SEARCH_QUERY];
-    const marker = adParams[AdParamKeys.MARKER];
+
+    if (searchQuery == undefined) {
+      this.simidProtocol.reject(eventData, {errorCode: CreativeErrorCode.UNSPECIFIED, 
+        message: "Required field 'SearchQuery' not found"});
```

Comment:
Consider making this searchQuery (camelcased) as opposed to SearchQuery bc the parameter is case sensitive and the error message might be misleading as to how to name it.

Or better yet just use AdParamKeys.SEARCH_QUERY here as a string template so that if we do decide to change the parameter name, we don't have to change a whole lot and just the declaration at the top of the file.

Reply:
switched to AdParamKeys.SEARCH_QUERY